### DISCUSSION TITLE
MWP Tools on Mac

### DISCUSSION THREAD

**Discussion by timtuxworth** [stronnag/mwptools#135] [Q&A]
I'm trying to build MWP tools on Mac OS (Ventura) and it failes with a dependency on libsoup 2.4. I have libsoup 2.74.2_1 and if I used home-brew it installed 3.2.2. There doesn't seem to be a way to get 2.4.

I'm thinking the easiest way would be to update the dependency to 2.74, how would I do this?

**Comment by stronnag** [comment]
I'd be interested to see if this works. I managed to compile mwp in a OSX VM some time ago, but it's not usable because GTK and OSX fail to agree a common encoding for OpenGL which is required by the map layer. So sadly it's unusable. Maybe this would not be a problem on real hardware. 

The `libsoup` (actually `libchamplain` [#128](https://github.com/stronnag/mwptools/discussions/128#discussioncomment-4739370) pain) is a problem, I'll see what I did about that for my OSX build.

**Reply by timtuxworth** [reply]
```
meson build --buildtype=release --strip --prefix $HOME/.local
The Meson build system
Version: 1.0.0
Source dir: /Users/tim/ArduPilot/mwptools
Build dir: /Users/tim/ArduPilot/mwptools/build
Build type: native build
Project name: mwptools
Project version: 6.019.690
C compiler for the host machine: cc (clang 14.0.0 "Apple clang version 14.0.0 (clang-1400.0.29.202)")
C linker for the host machine: cc ld64 820.1
Vala compiler for the host machine: valac (valac 0.56.3)
Host machine cpu family: aarch64
Host machine cpu: aarch64
Configuring mwp-config.h using configuration
Found pkg-config: /opt/homebrew/bin/pkg-config (0.29.2)
Run-time dependency glib-2.0 found: YES 2.74.5
Dependency glib-2.0 found: YES 2.74.5 (cached)
Did not find CMake 'cmake'
Found CMake: NO
Run-time dependency libsoup-2.4 found: NO (tried pkgconfig and framework)

meson/meson.build:55:2: ERROR: Dependency "libsoup-2.4" not found, tried pkgconfig and framework
```

**Reply by stronnag** [reply]
brew issue. It should install pkg-config files for libsoup-3.0 and libsoup-2.4 (based on 2.74, which is what Linux / FreeBSD does).

**Reply by timtuxworth** [reply]
Yes - I have both libsoup-3.0 and libsoup-2.74

**Comment by stronnag** [comment]
but brew fails to install `/usr/local/lib/pkgconfig/libsoup-2.4.pc` for `libsoup-2.74`. Which it should. Game over.
